packages/rich-text/src/plugins/Hr/index.tsx

```diff
@@ -55,19 +58,17 @@ interface ToolbarHrButtonProps {
 
 export function withHrEvents(editor: SPEditor) {
   return (event) => {
-    if (!editor.selection) return;
-
     const isEnter = event.keyCode === 13;
-    const [currentHrNode] = getNodes(editor, {
-      at: editor.selection.focus.path,
-      match: (node) => isElement(node) && node.type === BLOCKS.HR,
-    });
-    const isCurrentBlockHr = !!currentHrNode;
-
-    if (isEnter && isCurrentBlockHr) {
-      event.preventDefault();
-
-      Transforms.move(editor, { distance: 1 });
+    const [, pathToSelectedHr] = getNodeEntryFromSelection(editor, BLOCKS.HR);
+
+    if (pathToSelectedHr) {
+      if (isEnter) {
+        event.preventDefault();
+        moveToTheNextLine(editor);
+      } else if (event.key === 'Backspace') {
+        event.preventDefault();
+        Transforms.removeNodes(editor, { at: pathToSelectedHr });
+      }
```

Comment:
This is more than a refactor, the second conditional lets us delete a selected HR (which we couldn't before, we could only delete an HR by backspacing from the paragraph after it). This felt weird but more importantly it's inconsistent with behavior in the legacy editor.